framework/src/problems/DisplacedProblem.C

```diff
@@ -617,9 +617,11 @@ DisplacedProblem::reinitElem(const Elem * elem, THREAD_ID tid)
 void
 DisplacedProblem::reinitElemPhys(const Elem * elem,
                                  const std::vector<Point> & phys_points_in_elem,
-                                 THREAD_ID tid,
-                                 bool)
+                                 THREAD_ID tid)
 {
+  mooseAssert(_mesh.getMesh().query_elem_ptr(elem->id()) == elem,
+              "Are you calling this method with a undisplaced mesh element?");
```

Comment:
Beautiful assertions 🙌🏼 

FYI in the future for even less code, we have the `MooseMesh::queryElemPtr()` API